includes/Core/User_Input/User_Input.php

```diff
@@ -0,0 +1,314 @@
+	}
+
+	/**
+	 * Registers functionality.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		$this->site_specific_questions->register();
+		$this->user_specific_questions->register();
+
+		add_filter(
+			'googlesitekit_rest_routes',
+			function( $routes ) {
+				return array_merge( $routes, $this->get_rest_routes() );
+			}
+		);
+	}
+
+	/**
+	 * Gets related REST routes.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of REST_Route objects.
+	 */
+	private function get_rest_routes() {
+		$can_authenticate = function() {
+			return current_user_can( Permissions::AUTHENTICATE );
+		};
+
+		return array(
+			new REST_Route(
+				'core/user/data/user-input-settings',
+				array(
+					array(
+						'methods'             => WP_REST_Server::READABLE,
+						'callback'            => function() {
+							return rest_ensure_response( $this->get_answers() );
+						},
+						'permission_callback' => $can_authenticate,
+					),
+					array(
+						'methods'             => WP_REST_Server::CREATABLE,
+						'callback'            => function( WP_REST_Request $request ) {
+							$data = $request->get_param( 'data' );
+
+							if ( ! isset( $data['settings'] ) || ! is_array( $data['settings'] ) ) {
+								return new WP_Error(
+									'rest_missing_callback_param',
+									__( 'Missing settings data.', 'google-site-kit' ),
+									array( 'status' => 400 )
+								);
+							}
+
+							return rest_ensure_response(
+								$this->set_answers(
+									$data['settings']
+								)
+							);
+						},
+						'permission_callback' => $can_authenticate,
+						'args'                => array(
+							'data' => array(
+								'type'       => 'object',
+								'required'   => true,
+								'properties' => array(
+									'settings' => array(
+										'type'      => 'object',
+										'required'  => true,
+										'questions' => array_fill_keys(
+											array_keys( static::$questions ),
+											array(
+												'type'  => 'array',
+												'items' => array( 'type' => 'string' ),
+											)
+										),
+									),
+								),
+							),
+						),
+					),
+				)
+			),
+		);
+	}
+
+	/**
+	 * Gets the set of user input questions.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array The user input questions.
+	 */
+	public static function get_questions() {
+		return static::$questions;
+	}
+
+	/**
+	 * Gets user input answers.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array|WP_Error User input answers.
+	 */
+	public function get_answers() {
+		$questions = static::$questions;
+		$settings  = array_merge(
+			$this->site_specific_questions->get(),
+			$this->user_specific_questions->get()
+		);
+
+		// If there are no settings, return default empty values.
+		if ( empty( $settings ) ) {
+			array_walk(
+				$questions,
+				function ( &$question ) {
+					$question['values'] = array();
+				}
+			);
+
+			return $questions;
+		}
+
+		$user_id = get_current_user_id();
```

Comment:
It might be better to source this from `user_options->get_user_id()` instead since that is essentially the "effective user" this class is referring to.

Reply:
Updated, thanks!